management-app/src/main/java/com/nearsoft/fgaribay/mgmt/BrokerConnector.java

```diff
@@ -8,26 +8,28 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
-import java.util.Collections;
-import java.util.List;
+import java.util.ArrayList;
 
 @Component
 public class BrokerConnector {
 
     private final RabbitTemplate rabbitTemplate;
 
-    private ProductProperties.Queues queueProperties;
+    private ProductProperties.Exchanges exchangeProperties;
+    private ProductProperties.RoutingKeys routingKeyProperties;
 
     @Autowired
     public BrokerConnector(RabbitTemplate rabbitTemplate, ProductProperties properties) {
         this.rabbitTemplate = rabbitTemplate;
-        this.queueProperties = properties.getQueues();
+        this.exchangeProperties = properties.getExchanges();
+        this.routingKeyProperties = properties.getRoutingKeys();
     }
 
-    public List<Product> retrieveProductList() {
+    public ArrayList<Product> retrieveProductList() {
```

Comment:
Why did you decide this should be an ArrayList instead of a List?

Reply:
Thanks, I originally thought it was necessary for serialization. I have fixed it now to use List.